recipes/tz/all/conandata.yml

```diff
@@ -1,4 +1,8 @@
 sources:
   "2023c":
-    url: "https://github.com/eggert/tz/archive/refs/tags/2023c.tar.gz"
-    sha256: "9aa20ef838183e58f09acca92098cf6aa6d8e229aecf24e098c3af2a38e596f8"
+    "sources":
+      url: "https://github.com/eggert/tz/archive/refs/tags/2023c.tar.gz"
+      sha256: "9aa20ef838183e58f09acca92098cf6aa6d8e229aecf24e098c3af2a38e596f8"
+    "windows_zones":
+      url: "https://raw.githubusercontent.com/unicode-org/cldr/fd8ca965e03ca3cf07a0487678e0a1e002646ec0/common/supplemental/windowsZones.xml"
+      sha256: "e5720d0d82a0a62687184bf0d9472f3cf0f7ef8b930ad40f3ae7386d8cb57213"
```

Comment:
This is something that the `date` library expects to be present in the source database on windows. It contains the mapping between the windows time zone names and IANA time zones names ([description](https://stackoverflow.com/a/17348822))

You can see the dependency here:

https://github.com/HowardHinnant/date/blob/88a3b1512661decb78c60841f819a14eb200423c/src/tz.cpp#L661-L668

https://github.com/HowardHinnant/date/blob/88a3b1512661decb78c60841f819a14eb200423c/src/tz.cpp#L3686-L3688

If this file isn't in the same directory, it throws having failed to locate it.

Reply:
Note: CLDR is the defacto authority for these kinds of files - see more about the project here: https://cldr.unicode.org/index

Reply:
It's a "practically required" file to make the IANA tzdb useful on windows - without it, my understanding is that there isn't a way to convert between the windows timezones and IANA timezones. In that context, it could be viewed as a compatibility patch to enable windows compatibility for the package.

Having said that, if you'd prefer it in a different package I can attempt to do that. Please confirm if that's the way you would like me to go. My thought was that they fit together quite naturally in that a user who wants one would invariably want the other.

With respect to your comments about date: I didn't get the feeling that these changes wouldn't be supported out of principle, more time/maintenance constraints. The only modification required to date was to support reading the location of the database from an environment variable, which was a trivial modification.

Because the `date` library wasn't constructed with package management use in mind, external database provision wasn't a consideration and so the library makes some assumptions about what files are where, and also attempts to download its own files. Understandably this requires some modification to extricate these dependencies from each other.

Your comment gives me some doubt that this approach will be accepted on CCI though. If that's the case - please let me know so that I don't waste any more time pursuing this.

For what it's worth, this solution works really well for me locally. I can't imagine there's too much additional working in getting this functional, but I haven't been able to thoroughly test without the CI service whilst I'm blocked on the hook in https://github.com/conan-io/hooks/pull/525

Reply:
Another reason to consider bundling `windowsZones.xml` with the tz database is that it's often updated explicitly to support changes in tzdb - they are tightly coupled: https://github.com/unicode-org/cldr/commits/main/common/supplemental/windowsZones.xml

Reply:
@samuel-emrys thanks a lot for the useful links and clear explanation. This has helped me to understand how tz database from IANA, the CLDR, and the "date" library work together.

While I understand that having the windowsZones.xml file inside this tz database package is quite practical and convenient, I believe that we would create a "mix" package from CLDR and IANA sources which do not follow the original sources of the packages and could cause different expectations in the user of the recipe.

As described at https://howardhinnant.github.io/date/tz.html#Installation in "windows specfic", this WindowsZones.xml (although tightly coupled with tz database) is something that the **date** library expects to be installed *externally* at a certain location. It is not assumed as part of the tz database and not provided automatically by the date library AFAIK.

From my point of view, there should be a separate CLDR recipe that provides the needed file and that should be a require in the **date** recipe, leaving this tz recipe uncoupled with the WindowsZones.xml file. The downside of this is that the version of both tz recipe and the new CLDR (windowsZones.xml) will have to be compatible according to the historic of changes you pointed out here https://github.com/unicode-org/cldr/commits/main/common/supplemental/windowsZones.xml, however, I believe this dependency represents how the different pieces work together and would be closer to user expectations when using these libraries.

I hope that this reasoning makes sense. I would be happy to collaborate on this if needed. Thanks a lot for the hard work @samuel-emrys and thanks again for taking the time to explain this situation 😄

Reply:
Thanks for the clarification @danimtb, I'll get started on a `windowszones` package as well. For awareness because I don't think this is front of mind at the moment, this will mean additional changes to the `date` library, as it currently assumes that `windowsZones.xml` is present in the `tzdata` directory - it places it alongside the timezone database. This obviously won't be possible with the conan architecture as packages are immutable, so modifications will be required to decouple these.

Reply:
thanks for your understanding. Let's see if we can point the date library to pick the windows zones from a different location. Please ping me if you need any help on this.